packages/react-signature-frame/src/index.js

```diff
@@ -17,9 +17,26 @@ class AnvilSignatureFrame extends React.Component {
     window.removeEventListener('message', this.handleSignFinish)
   }
 
-  handleSignFinish = ({ origin, data: url }) => {
+  handleSignFinish = ({ origin, data }) => {
     if (this.props.anvilURL !== origin) return
-    this.props.onFinish(url)
+    if (typeof data === 'string') {
+      this.props.onFinish(data)
+
+      // parse query params into an object
+      const searchStr = data.split('?')[1]
+      const searchObj = JSON.parse('{"' + decodeURI(searchStr).replace(/"/g, '\\"').replace(/&/g, '","').replace(/=/g, '":"') + '"}')
+      const { signerStatus, signerEid, nextSignerEid, documentGroupStatus, documentGroupEid, etchPacketEid, weldDataEid } = searchObj
+      this.props.onFinishSigning({
```

Comment:
Can you also make sure there is no `error` or `errorType` in the `searchObj` before calling this? This callback should only be called on signer success. I'll add an error handler mirroring this in the next pass